quipucords/scanner/network/inspect_callback.py

```diff
@@ -1,239 +1,298 @@
-        event_data = event_dict.get("event_data")
-        host = event_data.get("host", UNKNOWN)
-        result = event_data.get("res")
-        task_action = event_data.get("task_action")
+    def task_on_ok(self, event_dict: dict):
+        """Handle successful ansible events."""
+        # example event
+        # {
+        #   "uuid": "9d097760-4d2c-43e5-8aba-d1a0e825aed7",
+        #   "counter": 140,
+        #   "stdout": "ok: [127.0.0.1] => <...>",
+        #   "start_line": 395,
+        #   "end_line": 401,
+        #   "runner_ident": "84418d26-0363-4924-92f2-d9f545f2a8ef",
+        #   "event": "runner_on_ok",
+        #   "pid": 84005,
+        #   "created": "2024-02-02T12:25:42.424553",
+        #   "parent_uuid": "6c946626-0f07-0f23-a006-000000000038",
+        #   "event_data": {
+        #     "playbook": "<...>",
+        #     "playbook_uuid": "470be249-5d66-43f8-a768-8bbfa7446f7b",
+        #     "play": "group_0",
+        #     "play_uuid": "6c946626-0f07-0f23-a006-000000000002",
+        #     "play_pattern": " group_0 ",
+        #     "task": "set internal_have_rpm",
+        #     "task_uuid": "6c946626-0f07-0f23-a006-000000000038",
+        #     "task_action": "set_fact",
+        #     "resolved_action": "ansible.builtin.set_fact",
+        #     "task_args": "",
+        #     "task_path": "<...>",
+        #     "role": "check_dependencies",
+        #     "host": "127.0.0.1",
+        #     "remote_addr": "127.0.0.1",
+        #     "res": {
+        #       "ansible_facts": {
+        #         "internal_have_rpm": true
+        #       },
+        #       "_ansible_no_log": null,
+        #       "changed": false
+        #     },
+        #     "start": "2024-02-02T12:25:42.412059",
+        #     "end": "2024-02-02T12:25:42.424405",
+        #     "duration": 0.012346,
+        #     "event_loop": null,
+        #     "uuid": "9d097760-4d2c-43e5-8aba-d1a0e825aed7"
+        #   }
+        # }
+        event_data = event_dict.get("event_data", {})
+        ansible_facts = event_data.get("res", {}).get("ansible_facts", {})
+        if not ansible_facts:
+            return
+        host = event_data.get("host", UNKNOWN_HOST)
         task = event_data.get("task")
-        # Print Role started for each host
-        if task_action == "set_fact" and task == STARTED_PROCESSING_ROLE:
-            log_message = f"PROCESSING {host} - ANSIBLE ROLE {self.last_role}"
-            self.scan_task.log_message(log_message)
-        task_facts = result.get("ansible_facts")
-        if task_facts:
-            self.process_task_facts(task_facts, host)
+        role = event_data.get("role")
+        logger.info("[host=%s] processing role=%s task=%s", host, role, task)
+        self._process_task_facts(ansible_facts, host)
+
+    def _process_task_facts(self, ansible_facts, host):
+        for fact_name, _fact_value in ansible_facts.items():
+            fact_value = sanitize_for_utf8_compatibility(_fact_value)
```

Comment:
this is change of order compared with the original implementation. sanitization was happening after post processing.

I decided to do that as early as possible to be more cautious.